src/components/Match/MatchContainer.tsx

```diff
@@ -35,13 +38,44 @@ import tryToReplaceWithLocalTaxon from "./helpers/tryToReplaceWithLocalTaxon";
 import Match from "./Match";
 import PreMatchLoadingScreen from "./PreMatchLoadingScreen";
 
-const setQueryKey = ( selectedPhotoUri, shouldUseEvidenceLocation ) => [
+interface ImageParamsType {
+  uri?: string;
+  image: {
+    uri: string;
+  };
+  lat?: number;
+  lng?: number;
+}
+
+interface NavParams {
+  id?: number | string;
+  firstPhotoID?: number | string;
+  representativePhoto?: { isRepresentativeButOtherTaxon?: boolean; id?: number | string };
+}
+
+interface StateType {
+  onlineFetchStatus: string;
```

Comment:
More like mutually exclusive states in the form of a few strings and not the general string type. So, maybe you want to refactor to an enum.